### PR TITLE
Support GCC 10

### PR DESCRIPTION
**Pull request by runlevel5**
* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open Pull Requests for the same change?
* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes? Here's an example.
* [x] Have you successfully run brew style with your changes locally?
* [x] Have you successfully run brew tests with your changes locally?
----

Fedora 32 Linux has been released with GCC10 despite of the fact that GCC 10 has not officially released. To help unblock Fedora 32 users, this PR would enable Fedora's GCC 10 to be picked up correctly.

Addressed https://github.com/Homebrew/brew/issues/7330

cc @Bo98

### CODE REVIEW

File: Library/Homebrew/development_tools.rb [hunk 1]
```diff
@@ -78,7 +78,7 @@ def non_apple_gcc_version(cc)
         path = HOMEBREW_PREFIX/"opt/gcc/bin"/cc
         path = locate(cc) unless path.exist?
         version = if path &&
-                     build_version = `#{path} --version`[/gcc(?:(?:-\d(?:\.\d)?)? \(.+\))? (\d\.\d\.\d)/, 1]
+                     build_version = `#{path} --version`[/gcc(?:(?:-\d(?:\.\d)?)? \(.+\))? (\d+\.\d\.\d)/, 1]
```
Comment (runlevel5) [inline Library/Homebrew/development_tools.rb:81]: FYI

```
$ gcc --version
gcc (GCC) 10.0.1 20200328 (Red Hat 10.0.1-0.11)
```

**Comment by Bo98** [comment]
We should also add support to the shims:

```
ln -s cc Library/Homebrew/shims/super/gcc-10
ln -s cc Library/Homebrew/shims/super/g++-10
ln -s cc Library/Homebrew/shims/mac/super/gcc-10
ln -s cc Library/Homebrew/shims/mac/super/g++-10
ln -s cc Library/Homebrew/shims/linux/super/gcc-10
ln -s cc Library/Homebrew/shims/linux/super/g++-10
```

**Comment by Bo98** [comment]
`Library/Homebrew/shims/linux/super/g++-10` appears to be missing from the latest commit

**Comment by runlevel5** [comment]
@Bo98 amended

**Comment by Bo98** [comment]
Some changes might be required here:

https://github.com/Homebrew/brew/blob/6b1d78ba7223858407d652ad73f23a86b0ea51f6/Library/Homebrew/shims/super/cc#L79-L87

`/\w\+\+(-\d+(\.\d)?)?$/` `/(g)?cc(-\d+(\.\d)?)?$/ `

**Comment by MikeMcQuaid** [comment]
@runlevel5 Mark this ready for review and we can merge. Also: please always use the PR template, thanks!

**Comment by runlevel5** [comment]
@MikeMcQuaid ready for review

**Comment by MikeMcQuaid** [comment]
Thanks again @runlevel5!

**Pull request merged**